runtime/src/mem/jemalloc/mem-jemalloc.c

```diff
@@ -28,15 +28,29 @@
 #include "chpl-comm.h"
 #include "chpl-mem.h"
 #include "chplmemtrack.h"
+#include "chplsys.h"
+#include "chpl-topo.h"
 #include "chpltypes.h"
 #include "error.h"
 
-static struct shared_heap {
+
+//
+// Information about shared heaps gotten from the comm layer.
+// MAX_HEAPS must be >= the number of sublocales.
+//
+#define MAX_HEAPS 4
```

Comment:
We can.  I'd need to use malloc(), of course, which would mean that we could never set up our jemalloc to intercept the C library allocator symbols, since here we'd need to do that malloc before we're set up.  Do we feel confident that we'll never want to do that?